### ISSUE TITLE
How to correctly use Javascript Registry with Bun?

### ISSUE THREAD

**Issue by jgay94** [jsr-io/jsr#325]
Hi all, excited with the prospect of JSR.

I'm trying to install an NPM package via the new [JSR][1] package registry with the following command: 

```bunx jsr add -D @jsr/total-typescript__ts-reset```

The command did generate a bunfig.toml, but did not install my dependency nor update my package.json file [like it's supposed to][2]. 

Here is the bunfig.toml that it created: 
```
[install.scopes]
"@jsr" = "https://npm.jsr.io"
```

And here is the error it throws: 
```
Setting up bunfig.toml...ok
Installing @total-typescript/ts-reset...
$ bun add --dev @total-typescript/ts-reset@npm:@jsr/total-typescript__ts-reset
bun add v1.0.35 (940448d6)
  🔍 @jsr/total-typescript__ts-reset [1/1]
error: package "@jsr/total-typescript__ts-reset" not found npm.jsr.io/@jsr%2ftotal-typescript__ts-reset 404
error: @total-typescript/ts-reset@npm:@jsr/total-typescript__ts-reset failed to resolve
Child process exited with: 1
```

Isn't JSR supposed to resolve to NPM via the ["@jsr scope"][3] if it can't find the package in the registry? Am I just executing the command incorrectly? 

Thank you!

  [1]: https://jsr.io/
  [2]: https://jsr.io/docs/using-packages
  [3]: https://youtu.be/dHfZiqVWVhk?si=9psWSZmDObNooq7A&t=505

**Comment by marvinhagemeister** [comment]
This is the expected behaviour. The way the JSR npm integration works is that it maps the `@jsr` scope to the JSR registry instead of going to `registry.npmjs.org`. So whenever you do `bun install @jsr/something` it will reach out to `npm.jsr.io`. It is not supposed to fall through to npm. That's not how overriding scopes work.

Checking the available packages on JSR it seems like a package called `@total-typescript/ts-reset` does indeed not exist: https://jsr.io/@total-typescript/ts-reset . I assume that you wanted to install the package from npm where it does exist: https://www.npmjs.com/package/@total-typescript/ts-reset .

To install npm packages in bun you should use the usual `bun add` command. The `bunx jsr add <package>` command is only needed when you want to install something from JSR and not from npm.

tl;dr:
```diff
- bunx jsr add -D @jsr/total-typescript__ts-reset
+ bun add -D @total-typescript/ts-reset
```

Explainer:
```sh
# Installing npm packages in bun
bun add <package>
# Installing JSR packages in bun
bunx jsr add <package>
```

**Comment by jgay94** [comment]
> This is the expected behaviour. The way the JSR npm integration works is that it maps the `@jsr` scope to the JSR registry instead of going to `registry.npmjs.org`. So whenever you do `bun install @jsr/something` it will reach out to `npm.jsr.io`. It is not supposed to fall through to npm. That's not how overriding scopes work.
> 
> Checking the available packages on JSR it seems like a package called `@total-typescript/ts-reset` does indeed not exist: https://jsr.io/@total-typescript/ts-reset . I assume that you wanted to install the package from npm where it does exist: https://www.npmjs.com/package/@total-typescript/ts-reset .
> 
> To install npm packages in bun you should use the usual `bun add` command. The `bunx jsr add <package>` command is only needed when you want to install something from JSR and not from npm.
> 
> tl;dr:
> 
> ```diff
> - bunx jsr add -D @jsr/total-typescript__ts-reset
> + bun add -D @total-typescript/ts-reset
> ```
> 
> Explainer:
> 
> ```shell
> # Installing npm packages in bun
> bun add <package>
> # Installing JSR packages in bun
> bunx jsr add <package>
> ```

Ah got it, I understand now. Thank you!!
